experiment/autobumper/bumper/bumper.go

```diff
@@ -39,13 +41,29 @@ const (
 	testImageRepo   = prowRepo
 )
 
-func Call(cmd string, args ...string) error {
+func Call(stdout, stderr io.Writer, cmd string, args ...string) error {
 	c := exec.Command(cmd, args...)
-	c.Stdout = os.Stdout
-	c.Stderr = os.Stderr
+	c.Stdout = stdout
+	c.Stderr = stderr
 	return c.Run()
 }
 
+type HideSecretsWriter struct {
+	Delegate    io.Writer
+	SecretAgent *secret.Agent
+}
+
+func (w HideSecretsWriter) Write(content []byte) (int, error) {
+	if w.SecretAgent == nil {
```

Comment:
Tried to avoid this pointer check by using the value, but hit [go-vet](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/14248/pull-test-infra-bazel/1171573224182386688#0:build-log.txt%3A4770) issue.

Any idea of workaround with the value?

Reply:
Probably no workaround: https://github.com/golang/go/issues/8005#issuecomment-190753527

Reply:
You could just make 

```go
type censorer interface {
    Censor([]byte) []byte
}

type HideSecretsWriter struct {
    censorer
}
```